test/src/main/java/org/apache/accumulo/test/zookeeper/ZooKeeperTestingServer.java

```diff
@@ -41,6 +46,31 @@
  */
 public class ZooKeeperTestingServer implements AutoCloseable {
 
+  public static class ZooKeeperForTests extends ZooKeeper {
+
+    public ZooKeeperForTests(String connectString, int sessionTimeout, Watcher watcher)
+        throws IOException {
+      super(connectString, sessionTimeout, watcher);
+    }
+
+    @Override
+    public void sync(final String path, VoidCallback cb, Object ctx) {
+      PathUtils.validatePath(path);
+
```

Comment:
I don't think it's a good idea to create a subclass that effectively changes the behavior of `sync` to diverge from the behavior documented in the API javadoc for the base class. I also don't think it's a good idea to call the ZK internals in here, which are not necessarily stable across versions. I have another idea, I'm going to put up an alternative PR that uses a custom callback that I think is simpler.